lib/_emerge/unmerge.py

```diff
@@ -563,7 +563,13 @@ def unmerge(root_config, myopts, unmerge_action,
 	maxval = len(all_selected)
 
 	for x in range(len(pkgmap)):
-		for y in pkgmap[x]["selected"]:
+
+		selected = pkgmap[x]["selected"]
+
+		if ordered:
+			selected = sorted(selected)
```

Comment:
You may need to use some Portage's helper function for comparisons of versions to avoid lexicographic comparison:
```
$ python -c 'print(sorted(["1.0", "1.1", "1.2", "1.9", "1.10", "1.11"]))'
['1.0', '1.1', '1.10', '1.11', '1.2', '1.9']
```